### ISSUE TITLE
JackTrip doesn't quit cleanly when still connected

### ISSUE THREAD

**Issue by ntonnaett** [jacktrip/jacktrip#552]
```
Waiting for Peer...
Received Connection from Peer!
Stopping JackTrip...
sending exit packet
JackTrip Processes STOPPED!
---------------------------------------------------------
corrupted double-linked list
Abgebrochen (Speicherabzug geschrieben)
```

I've never seen this before, but I'm not sure if this was introduced by virtualstudio mode. I used classic mode.

**Comment by ntonnaett** [comment]
I can't reproduce this with the xcb platform plugin right now. So it might be a wayland issue of Qt or just luck.

**Comment by psiborg112** [comment]
Yeah, after multiple attempts on windows and linux (not wayland) I can't trigger this.

**Comment by mikedickey** [comment]
I just tried several times on Fedora 35 (with Wayland) and was unable to reproduce.

**Comment by forsythac** [comment]
I just saw this behavior on Mac with a customer. needed to force quit the application when connected to a studio

**Comment by OConnorStP** [comment]
I have a Mac on which Jacktrip needs to be force-quit when connecting to a classic server that I can run diagnostics on if folks have suggestions.

**Comment by SynthiaPayne** [comment]
No issues whatsoever on my Catalina OSX with 1.6.0-rc.2, connecting in 
RtAudio or JACK to a Virtual Studio . Clean in and out.

synth

**Comment by OConnorStP** [comment]
I ran 1.5.3 and  1.6.0 rc3 to a Virtual Studio from a Mac running Monterey -- both hang on disconnect

I ran 1.5.1 and  1.6.0 rc3 to the same Virtual Studio from a Mac running Big Sur -- both worked

**Comment by forsythac** [comment]
Hey @OConnorStP Is JackTrip ALWAYS crashing on High Sierra? Is this known?
I think that should be its own issue if I'm not mistaken. Getting a support call from someone on high sierra that crashes on startup. If this isn't a supported version of macOS we should put that info in the docs.

**Comment by OConnorStP** [comment]
hi Andrew,

i just jumped on the High Sierra machine and reran that test (still on 1.6.0 rc3, not the latest).  it still crashed in GUI mode, but i switched to terminal and was successful at connecting.  Jacktrip started Jack (because it had been in RC Audio backend mode the last time it ran successfully??).  i don't know if this problem is widespread, my High Sierra users are all on summer break.

**Comment by forsythac** [comment]
Hey Mike, thanks for the double check. Any chance you can get the report from the crash when in gui mode? I'm filing a separate github issue for this.

**Comment by OConnorStP** [comment]
Cross posting back to this thread from Andrew's new ( #671 ) issue...  Aaron shared his build of the latest RC ( https://www.psi-borg.org/JackTrip.pkg ) which also solved the GUI-mode problem I described here.  It also runs fine in Terminal mode, but did throw this error on exit:

Waiting for Peer...
Received Connection from Peer!
^C
Shutting Down...
sending exit packet
Stopping JackTrip...
JackMachSemaphore::TimedWait name = js501.172.104.11.23 usec = 9223372036854775807 err = (os/kern) aborted
SuspendRefNum error
JackClient::Execute error name = 172.104.11.23
JackTrip Processes STOPPED!

**Comment by psiborg112** [comment]
What version of JACK and MacOS? There have been issues with Mach Semaphores of late, that the JACK developers have had to work around. (There was an issue that was causing JACK to crash on client disconnect in Monterey that has been fixed, but this seems to be an echo of those sorts of issues.)

**Comment by OConnorStP** [comment]
Jack is 1.9.21 (I just reinstalled it and ran the test again, same result).  I used the Universal binary installer.  OSX is 12.5.

Then I ran Jacktrip 1.6.1 on another M1 Mac that's running older versions of both Jack (1.9.20) and Monteray (10.2) and got a slightly different semaphore message:

Waiting for Peer...
Received Connection from Peer!
^C
Shutting Down...
sending exit packet
Stopping JackTrip...
JackTemporaryException : now quits...
Jack main caught signal 2
JackTrip Processes STOPPED!
---------------------------------------------------------
mikeoconnor@M1-Mini ~ % JackMachSemaphore::Destroy failed to kill semaphore server thread...
JackMachSemaphore::Destroy failed to kill semaphore server thread...
